### PR TITLE
Refuse to run if --check has not first been run successfully

### LINKED ISSUE #333: refuse to run elevate-cpanel if --check has not first been run successfully.
We should be able to check the blocker file for whether the last run was good.

### PR DESCRIPTION
**Pull request by timmullin**
By submitting pull requests to this repo, I agree to the Contributor License Agreement which can be found at: https://github.com/cpanel/elevate/blob/main/docs/cPanel-CLA.pdf

Fixes #333

### CODE REVIEW

File: script/elevate-cpanel.PL [hunk 1]
```diff
@@ -515,6 +515,7 @@ sub monitor_upgrade ($self) {
 
 sub start ($self) {
     Elevate::Blockers::Distros::bail_out_on_inappropriate_distro();
+    Elevate::Blockers::bail_if_check_not_run_or_blockers_found();
```
Comment (toddr) [inline script/elevate-cpanel.PL:518]: does this execute if I run `--check`?

Comment (toddr) [reply script/elevate-cpanel.PL:518]: no it doesn't. nm

File: lib/Elevate/Blockers.pm [hunk 2]
```diff
@@ -112,6 +113,50 @@ sub check ( $self, %opts ) {    # do_check - main  entry point
...
+    }
+
+    FATAL("You must fix all blockers and re-run with --check before performing the upgrade.");
+    exit 1;
+}
+
+sub check_run_recently () {
+    return 0 if ( !-f ELEVATE_BLOCKER_FILE );
+
+    return ( ( stat(ELEVATE_BLOCKER_FILE) )[9] + ELEVATE_BLOCKER_FILE_EXPIRE_TIME > time() ) ? 1 : 0;
```
Comment (toddr) [inline lib/Elevate/Blockers.pm:142]: You should be able to test this with Test::MockFile. From my cPanel blocker experience, we often got this logic wrong :(

Comment (timmullin) [reply lib/Elevate/Blockers.pm:142]: It didn't look like Elevate::Blockers had a unit test; I guess I could start one & test the functions I added in it.

**Pull request merged**